packages/components/src/components/Tooltip/TooltipFloatingUi.tsx

```diff
@@ -0,0 +1,213 @@
+import {
+    useFloating,
+    autoUpdate,
+    offset,
+    flip,
+    shift,
+    useHover,
+    useFocus,
+    useDismiss,
+    useRole,
+    useInteractions,
+    FloatingPortal,
+    useDelayGroup,
+    useDelayGroupContext,
+    useMergeRefs,
+    useTransitionStyles,
+    arrow,
+} from '@floating-ui/react';
+import type { Placement, UseFloatingReturn } from '@floating-ui/react';
+import {
+    useState,
+    useMemo,
+    createContext,
+    useContext,
+    ReactNode,
+    HTMLProps,
+    forwardRef,
+    isValidElement,
+    cloneElement,
+    useRef,
+    RefObject,
+    CSSProperties,
+    MutableRefObject,
+} from 'react';
+
+/**
+ * Based on https://floating-ui.com/docs/tooltip but heavily modified
+ */
+
+type ArrowRef = RefObject<SVGSVGElement>;
+
+interface TooltipOptions {
+    initialOpen?: boolean;
+    placement?: Placement;
+    open?: boolean;
+    onOpenChange?: (open: boolean) => void;
+    offset?: number;
+}
+
+type UseTooltipReturn = ReturnType<typeof useInteractions> & {
+    open: boolean;
+    setOpen: (open: boolean) => void;
+    arrowRef: RefObject<SVGSVGElement>;
+} & UseFloatingReturn;
+
+export const useTooltip = ({
+    initialOpen = false,
+    placement = 'top',
+    open: controlledOpen,
+    onOpenChange: setControlledOpen,
+    offset: offsetValue = 10,
+}: TooltipOptions = {}): UseTooltipReturn => {
+    const arrowRef = useRef<SVGSVGElement>(null);
+    const [uncontrolledOpen, setUncontrolledOpen] = useState(initialOpen);
```

Comment:
nitpick: This name seems to me a bit unclear. I would change it to `isUncontrolledTooltipOpen` or something like it. You have context here, but for others it might be confusing.

Reply:
fixed in a640d593f5